src/types/field/baseField.ts

```diff
@@ -1,32 +1,20 @@
 import { Document } from 'mongoose'
 
+import { AllowMyInfoBase, FieldBase } from '../../../shared/types/field'
 import { IFormSchema } from '../form'
 
-import { BasicField, MyInfoAttribute } from './fieldTypes'
-
-export interface IMyInfo {
-  attr: MyInfoAttribute
-}
-
+export type IMyInfo = NonNullable<AllowMyInfoBase['myInfo']>
 export interface IMyInfoSchema extends IMyInfo, Document {
   /** Returns the top level document of this sub-document. */
   ownerDocument(): IFormSchema
   /** Returns this sub-documents parent document. */
   parent(): IFieldSchema
 }
 
-export interface IField {
-  globalId?: string
-  title: string
-  description: string
-  required: boolean
-  disabled: boolean
-  fieldType: BasicField
-  myInfo?: IMyInfo
-}
+export type IField = FieldBase
```

Comment:
is it really necessary to reexport all these types as aliases of the `Base` type? also haven't we always been strict about only naming interfaces starting with `I` previously?

Reply:
I did not want to rename everything in the backend whilst doing this refactor. There should hopefully be a final PR where all these aliases are removed and only the shared names are used. I can also do it in this PR if needed

Reply:
ah ok sounds good!

Reply:
separate PR is good, as long as we do it eventually